js/offline-form.js

```diff
@@ -0,0 +1,115 @@
+/* global ERROR_OFFLINE_URL, ERROR_MESSAGES, ERROR_500_URL */
+
+// IIFE is used for lexical scoping instead of just a braces block due to bug with const in Safari.
+(() => {
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'wpPendingForm' );
+	const errorMessages = {
+		"serverOffline": "The server appears to be down. Please try again later.",
+		"error"        : "Something prevented the page from being rendered. Please try again.",
+		"form"         : "Your form will be submitted once you are back online!"
```

Comment:
These error message should be added in PHP and localize in JS and make strings translatable.